### PR TITLE
VAULT-15546 First pass at Vault Proxy docs

### PR DESCRIPTION
**Pull request by VioletHynes**
This should be a complete set of the docs updates for the Vault Agent use case split into Vault Agent and Vault Proxy. I opted to make a meta-category for both since they share subsystems like auto-auth. I kept caching specific to Proxy/Agent, even though the pages are very similar today, as there can be differences in behaviour going forward.

For context for reviewers that might not be aware: we're splitting Vault Agent into Vault Agent and Vault Proxy. Proxy will contain the API proxy functionality of Agent, and we plan to remove Vault Agent's API proxy functionality in a later release.

High level summary of the docs changes:
- Added new top level navigation category, `Vault Agent and Vault Proxy`, where Vault Agent was
- `Vault Agent and Vault Proxy`'s section contains the high level "why should I use either" information
- Vault Agent's top level navigation category is moved under `Vault Agent and Vault Proxy`, with the index changed slightly
- References to Agent's API proxy have not been removed (yet), but a deprecation notice has been added
- The `Auto-Auth` section is moved under `Vault Agent and Vault Proxy`, and language updated to reflect that it applies to both Agent and Proxy
- New section added, `Vault Proxy`, containing a similar set of pages as the Agent documentation -- mostly copied over with salient parts changed

🔍 [Deploy preview](https://vault-git-violethynes-vault-15546-hashicorp.vercel.app/vault/docs/agent-and-proxy)

I've updated the redirects to avoid breaking existing links to Agent documentation. It seems to work for all of the pages!

### CODE REVIEW

File: website/content/api-docs/index.mdx
Comment (VioletHynes) [inline website/content/api-docs/index.mdx:1]: Apologies for the whitespace changes on this file, though it seems to be pretty much only affecting this file. The only salient change is at the bottom, line 317.

**Comment by stevendpclark** [comment]
Hi Violet, will this not break any bookmarked links, outside references to all the existing agent pages? Is that something we want to do?

**Comment by VioletHynes** [comment]
> Hi Violet, will this not break any bookmarked links, outside references to all the existing agent pages? Is that something we want to do?

Hi there, and yeah, you're right that this PR will have this impact. Ultimately, I don't see a way to avoid this without duplicating a lot of existing pages. For example, auto-auth is something that's no longer Agent specific, so unless we move it out of the Agent category, we'd need to duplicate it in a new category, or it would appear mischaracterized. 

Furthermore, the index of today's Agent page talks about why a customer might need Agent/Proxy - it makes a lot of sense to me for that kind of discussion to be in a meta-category. I guess, to put it another way, I feel like there's no good way to preserve existing links while maintaining well-organized docs, though it's possible I've missed something (or that others might feel different).

I've already looped in the Education team for review on this PR, as this is a larger reaching docs PR than normal, and am planning to review with them first. Hopefully, we can determine whether or not this new structure is the right path forward!

**Comment by stevendpclark** [comment]
Thanks for the explanation @VioletHynes, makes perfect sense and I agree the new layout is better going forward. I think we could possibly leverage a [Vercel redirect](https://vercel.com/docs/concepts/edge-network/redirects) to cover the majority of these pages without having to duplicate the docs. 

Within [website/redirects.js](https://github.com/hashicorp/vault/blob/main/website/redirects.js) I added this and it *seemed* to work, I haven't checked if this covers all pages 
```
{
  source: '/vault/docs/agent/:slug',
  destination: '/vault/docs/agentandproxy/agent/:slug',
  permanent: true,
}
```

**Comment by VioletHynes** [comment]
> Thanks for the explanation @VioletHynes, makes perfect sense and I agree the new layout is better going forward. I think we could possibly leverage a [Vercel redirect](https://vercel.com/docs/concepts/edge-network/redirects) to cover the majority of these pages without having to duplicate the docs.
> 
> Within [website/redirects.js](https://github.com/hashicorp/vault/blob/main/website/redirects.js) I added this and it _seemed_ to work, I haven't checked if this covers all pages
> 
> ```
> {
>   source: '/vault/docs/agent/:slug',
>   destination: '/vault/docs/agentandproxy/agent/:slug',
>   permanent: true,
> }
> ```

Oh this is great! Thanks for calling this out, I'll give this a go.

**Comment by VioletHynes** [comment]
Updated the PR with redirection! I needed to be a bit more explicit with the redirection since not all `agent` stuff moved to `agentandproxy/agent` (e.g. auto-auth). Thanks a lot @stevendpclark for the suggestion!

File: website/content/docs/agentandproxy/autoauth/sinks/index.mdx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+layout: docs
+page_title: Vault Agent and Vault Proxy Auto-Auth Sinks
+description: Sinks for Auto-Auth
+---
+
+# Vault Agent Auto-Auth Sinks
+
+Every time an auto-auth authentication is successful, the token is written to the
+configured Sinks, subject to their configuration. Today, we only support one
```
Comment (brianshumate) [inline website/content/docs/agentandproxy/autoauth/sinks/index.mdx:10]:
```suggestion
enabled Sinks, subject to their configuration. Today, we only support one
```

**Review by brianshumate — CHANGES_REQUESTED** [review]
This all makes sense to me and looks great so far. I have only a couple minor suggestions!

File: website/content/docs/agentandproxy/autoauth/methods/cert.mdx [hunk 2]
```diff
@@ -13,8 +13,8 @@ to use certificates which differ from those used in the `vault` stanza.
 It is strongly advised to provide TLS settings in the configuration stanza
 within the auth method to avoid agent cache, if also enabled, from using the
 same TLS settings when proxying requests. If TLS settings are not present in the
-config stanza, Agent will fall back to using TLS settings from the [`vault`
-Stanza](/vault/docs/agent#vault-stanza).
+config stanza, Agent and Proxy will fall back to using TLS settings from thir respective
```
[the remark this answers is not in the document]
Comment (VioletHynes) [reply website/content/docs/agentandproxy/autoauth/methods/cert.mdx]: Good catch! Fixed, and pushed.

File: website/content/docs/agentandproxy/autoauth/sinks/index.mdx (hunk 1 above)
Comment (VioletHynes) [reply website/content/docs/agentandproxy/autoauth/sinks/index.mdx]: Good catch! Fixed, and pushed.

**Review by brianshumate — APPROVED** [review]

**Review by yhyakuna — COMMENTED** [review]
Great work!  I agree with the need for these changes. 

Please see https://github.com/hashicorp/vault/pull/20612 for my suggestions. If you don't agree with my suggestions, feel free to let me know or make changes.  

I wish I had a diagram for API Proxy, but I couldn't find one... :(

File: website/content/docs/agentandproxy/index.mdx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+- [Caching][caching] - Allows client-side caching of responses containing newly
+created tokens and responses containing leased secrets generated off of these
+newly created tokens. The agent also manages the renewals of the cached tokens and leases.
+
+## Auto-Auth
+
+[Vault Agent][vaultagent] and [Vault Proxy][vaultproxy] allow for easy authentication to Vault
+in a wide variety of environments. Please see the [Auto-Auth docs][autoauth] for information.
+
+Auto-Auth functionality takes place within an `auto_auth` configuration stanza.
```
Comment (jonathanfrappier) [inline website/content/docs/agentandproxy/index.mdx:117]: Is it worth providing an example here, or linking to an example of the configuration stanza?

**Review by yhyakuna — APPROVED** [review]

**Pull request merged**